batch/batch/worker/worker.py

```diff
@@ -844,12 +847,14 @@ async def delete_container(self):
             except Exception:
                 log.exception('while deleting container', exc_info=True)
 
-        try:
-            await check_shell(f'umount -l {self.container_overlay_path}/merged')
-        except asyncio.CancelledError:
-            raise
-        except Exception:
-            log.exception('while unmounting overlay', exc_info=True)
+        if self.overlay_mounted:
+            try:
+                await check_shell(f'umount -l {self.container_overlay_path}/merged')
+                self.overlay_mounted = False
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                log.exception('while unmounting overlay', exc_info=True)
```

Comment:
Add the container name here.

Reply:
Ah I have that in #10805. I can combine them though.